fieldsolver/derivatives.cpp

```diff
@@ -485,3 +485,137 @@ void calculateBVOLDerivativesSimple(
 
    phiprof::stop("Calculate volume derivatives",N_cells,"Spatial Cells");
 }
+
+/*! \brief Returns volumetric E of cell
+ *
+ */
+std::array<Real, 3> getE(SpatialCell* cell)
+{
+   return std::array<Real, 3> { {cell->parameters[CellParams::EXVOL], cell->parameters[CellParams::EYVOL], cell->parameters[CellParams::EZVOL]} };
+}
+
+/*! \brief Returns perturbated volumetric B of cell
+ *
+ */
+std::array<Real, 3> getPerB(SpatialCell* cell)
+{
+   return std::array<Real, 3> { {cell->parameters[CellParams::PERBXVOL], cell->parameters[CellParams::PERBYVOL], cell->parameters[CellParams::PERBZVOL]} };
+}
+
+/*! \brief Calculates momentum density of cell
+ *
+ */
+std::array<Real, 3> getP(SpatialCell* cell)
```

Comment:
Should these be ```static``` maybe? As they are not in a namespace, having something called getE might clash with things somewhere else in the code otherwise.

Reply:
Each is used a few times later on in this file, so I though it would be better not to repeat myself.

Reply:
getP can be confusing as in Vlasiator we usually use P for pressure (a second velocity moment)